packages/astro/src/runtime/client/dev-overlay/overlay.ts

```diff
@@ -418,24 +407,39 @@ export class AstroDevOverlay extends HTMLElement {
 	}
 
 	getPluginCanvasById(id: string) {
-		return this.shadowRoot.querySelector(`astro-dev-overlay-plugin-canvas[data-plugin-id="${id}"]`);
+		return this.shadowRoot.querySelector<HTMLElement>(
+			`astro-dev-overlay-plugin-canvas[data-plugin-id="${id}"]`
+		);
 	}
 
-	togglePluginStatus(plugin: DevOverlayPlugin, status?: boolean) {
+	async togglePluginStatus(plugin: DevOverlayPlugin, status?: boolean) {
```

Comment:
nit: `status` as a boolean is very cryptic. For example what `status = false` mean? Maybe we should choose a different name

Reply:
would `pluginStatus` be a better name for you? Since it's a parameter of `togglePluginStatus`, I feel like it was pretty clear like this.

`status` matches the other API that toggle the plugin's state as well, but it's always done in the context of the plugin, so it felt fine to me

Reply:
Oh, Ok, I understand now. Maybe `hasNewStatus`?

Reply:
I changed it to `newStatus`!